### PR TITLE
Increase stop impact on sharp turns from ramps and turn channels.

### LINKED ISSUE #647: Should turn left at intersection instead of bear right and then sharp left
http://www.openstreetmap.org/directions?engine=mapzen_car&route=40.31250%2C-75.10961%3B40.31277%2C-75.10757#map=19/40.31231/-75.10859

screenshot from 2017-04-11 08-30-26

### PR DESCRIPTION
**Pull request by dnesbitt61**
Fixes #647

### CODE REVIEW

File: src/mjolnir/graphenhancer.cc [hunk 1]
```diff
@@ -723,9 +723,22 @@ uint32_t GetStopImpact(uint32_t from, uint32_t to,
...
+  bool is_sharp = (turn_type == Turn::Type::kSharpLeft ||
+                   turn_type == Turn::Type::kSharpRight ||
+                   turn_type == Turn::Type::kReverse);
+  bool is_slight = (turn_type == Turn::Type::kStraight ||
+                    turn_type == Turn::Type::kSlightRight ||
+                    turn_type == Turn::Type::kSlightLeft);
   if (allramps) {
-    stop_impact /= 2;
+    if (is_sharp) {
+      stop_impact + 2;
```
Comment (dgearhart) [inline src/mjolnir/graphenhancer.cc]: should be +=

File: src/mjolnir/graphenhancer.cc [hunk 2]
```diff
@@ -723,9 +723,22 @@ uint32_t GetStopImpact(uint32_t from, uint32_t to,
...
+    } else if (is_slight) {
+      stop_impact / 2;
```
Comment (dgearhart) [inline src/mjolnir/graphenhancer.cc]: /=

File: src/mjolnir/graphenhancer.cc [hunk 3]
```diff
@@ -736,12 +749,23 @@ uint32_t GetStopImpact(uint32_t from, uint32_t to,
     }
   } else if (edges[from].use() == Use::kRamp && edges[to].use() != Use::kRamp) {
     // Increase stop impact on merge
-    stop_impact += 2;
-  } else if (edges[from].use() == Use::kTurnChannel) {
-    if (edges[to].use() == Use::kRamp) {
+    if (is_sharp) {
+      stop_impact + 3;
```
Comment (dgearhart) [inline src/mjolnir/graphenhancer.cc]: +=

**Comment by dnesbitt61** [comment]
Thanks! Re-running route tests.

**Comment by dgearhart** [comment]
:shipit:

**Pull request merged**